lib/praxis/resource_definition.rb

```diff
@@ -48,6 +48,37 @@ def version(version=nil, options= { using: [:header,:params] }.freeze )
         @version_options = options
       end
 
+      def canonical_path( action_name=nil )
+        if action_name
+          raise "Canonical path for #{self.name} is already defined as: '#{@canonical_action_name}'. 'canonical_path' can only be defined once." if @canonical_action_name
+          @canonical_action_name = action_name
+        else
+          # Resolution of the actual action definition needs to be done lazily, since we can use the `canonical_path` stanza
+          # at the top of the resource, well before the actual action is defined.
+          unless @canonical_action
+            href_action = @canonical_action_name || DEFAULT_RESOURCE_HREF_ACTION
+            @canonical_action = actions.fetch(href_action) do
+              raise "Error: trying to set canonical_href of #{self.name}. Action '#{href_action}' does not exist"
+            end
+          end
+          return @canonical_action
+        end
+      end
+      
+      def to_href( params )
+        canonical_path.primary_route.path.expand(params)
+      end
+
+      def parse_href(path)
```

Comment:
well...I'm not sure it is equivalent when there are many other params defined (for the query string) and the fact of reporting "nulls" or not.
I'd say we can play it safe, and isolate from possible changes in the loading of the complete params Struct...and just load the one we know we need to...